src/Workspaces/Core/Portable/Editing/SyntaxEditor.cs

```diff
@@ -116,13 +155,14 @@ internal void ReplaceNode<TArgument>(SyntaxNode node, Func<SyntaxNode, SyntaxGen
         /// <param name="newNode">The new node that will be placed into the tree in the existing node's location.</param>
         public void ReplaceNode(SyntaxNode node, SyntaxNode newNode)
         {
-            CheckNodeInTree(node);
+            CheckNodeInOriginalTreeOrTracked(node);
             if (node == newNode)
             {
                 return;
             }
 
-            this.ReplaceNode(node, (n, g) => newNode);
+            newNode = ApplyTrackingToNewNode(newNode);
+            _changes.Add(new ReplaceChange(node, (n, g) => newNode, ApplyTrackingToNewNode));
```

Comment:
slightly not great that ths is an allocation per ApplyTrackingToNewNode.  consider computing the delegate once and passing that isntance along.